kubedb-postgres/Tiltfile

```diff
@@ -0,0 +1,10 @@
+
+is_installed = str(local('kubectl api-resources | grep postgres | wc -l')).strip()
+if is_installed == '0':
+  local('curl -fsSL https://raw.githubusercontent.com/kubedb/cli/0.12.0/hack/deploy/kubedb.sh | bash')
+
+k8s_yaml('postgres.yaml')
+k8s_resource(new_name='postgres',
+             objects=['quick-postgres'],
```

Comment:
I feel like the API here isn't quite right...but i'm also not totally sure what the API should be.

Reply:
What doesn't feel good about it? I just want to dig in more.

Reply:
if you have multiple custom resources for the same custom resource definition, you need a k8s_resource(new_name...) call for each one...it feels repetitive. 

You have to have a pretty advanced mental model for how tilt + k8s work together to use these functions well

i've been playing with a bunch of ideas for how you might make these apis compose together better. one idea was to use k8s annotations as the source of truth for a lot of what we use k8s_resource for now, e.g.,

```
kind: Postgres
metadata:
  name: quick-postgres
  annotations:
    tilt.dev/resource: postgres
    tilt.dev/pod-selector: "..."
```

and then k8s_resource would be syntactic sugar around setting the annotations (if you didn't want to put the annotations in the yaml)

if you did that, then you could have an api that mimics k8s admission controllers like:

```
def handlePostgres(obj):
  obj.metadata.resource['tilt.dev/resource'] = obj.metadata.name

k8s_admission_controller(kind='Postgres', handlePostgres)
```

and the APIs would compose together a bit better

but again, this is an api I'm still playing around with, and could use some fleshing out

Reply:
(and still glad that we have the object selector api, since it enables things like this and better exploration of the space)

Reply:
Yup, that makes sense. I'd love to try an annotations focused version of this API.